docs/guides/versioning.md

```diff
@@ -0,0 +1,214 @@
+    - [A `PropTypes.func` prop type is changed to have different arguments](#a-proptypesfunc-prop-type-is-changed-to-have-different-arguments)
+    - [A `PropTypes.func` prop type is changed to have additional arguments](#a-proptypesfunc-prop-type-is-changed-to-have-additional-arguments)
+    - [A `PropTypes.func` prop type is changed to have fewer arguments](#a-proptypesfunc-prop-type-is-changed-to-have-fewer-arguments)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- prettier-ignore-end -->
+
+## Overview
+
+The Carbon Design System team aims to follow
+[Semantic Versioning](https://semver.org/) (semver) for each of the packages
+that we ship. From semver.org, this means that:
+
+> Given a version number MAJOR.MINOR.PATCH, increment the:
+>
+> 1. **MAJOR** version when you make incompatible API changes,
+> 2. **MINOR** version when you add functionality in a backwards compatible
+>    manner, and
+> 3. **PATCH** version when you make backwards compatible bug fixes.
+>
+> _Additional labels for pre-release and build metadata are available as
+> extensions to the MAJOR.MINOR.PATCH format._
+
+As a result, whenever you see a `minor` or `patch` update for a package from the
+Carbon Design System you should feel confident that you can update without
+anything breaking in project.
+
+**Note:** If you ever bring in an update and it does break something in your
+project, please create an
+[issue](https://github.com/carbon-design-system/carbon/issues/new?assignees=&labels=type%3A+bug+%F0%9F%90%9B&template=bug-report.md&title=Breaking%20change)
+and we will resolve the issue as quickly as possible.
+
+In the following sections, you'll find specific details for our packages and the
+types of changes you can expect to occur as a consumer of that package. We'll
+try and highlight common changes in code and the corresponding `semver` bump
+that you can expect for the package when it is updated.
+
+## `carbon-components-react`
+
+### Changes
+
+| Type of change                                                      | semver bump |
+| ------------------------------------------------------------------- | ----------- |
+| A new prop is added to a component                                  | `minor`     |
+| An existing prop is deprecated                                      | `minor`     |
+| An existing prop is removed                                         | `major`     |
+| An existing prop type is changed to be more specific                | `major`     |
+| An existing prop type is changed to be more generic                 | `minor`     |
+| A `PropTypes.func` prop type is changed to have different arguments | `major`     |
+| A `PropTypes.func` prop type is changed to have additional argument | `minor`     |
+| A `PropTypes.func` prop type is changed to have fewer arguments     | `major`     |
+
+### Examples
+
+#### A new prop is added to a component
```

Comment:
```suggestion
#### A new prop is added to a component:
```

We've finally decided that we *are* doing colons after H4s

Reply:
As this isn't for the website, do we need this convention?